### ISSUE TITLE
Grad CAM Map With Memory Saving Looks Odd

### ISSUE THREAD

**Issue by gitrdonator** [cybertronai/gradient-checkpointing#40]
Has anyone tried to create grad cam or saliency maps after using `memory_saving_gradients`? Mine look very odd (columnar). I'm doing the Udacity program, and my `keras-vis` `grad-cam` plots look like this:
![cam_map_00](https://user-images.githubusercontent.com/44979155/60774640-9a2c9700-a106-11e9-8baf-69a9310abfa8.png)

**Comment by yaroslavvb** [comment]
I haven't....I can't really tell if this looks odd or not, the gradient results should be the same with and without memory saving

**Comment by gitrdonator** [comment]
@yaroslavvb I've noticed my model runs faster, with less memory, but with errors when I use the tensorflow implementation of the keras backend (`from tensorflow.python.keras._impl.keras import backend as K`, or `tensorflow.keras.backend` in later versions of tensorflow) instead of the backend with my keras install (`from keras import backend`). The only difference between my code and your keras test code (`keras_test.py`) is that I use the keras functional model API, e.g.

```
a = Input(shape=(32,))
b = Dense(32)(a)
model = Model(inputs=a, outputs=b)
```

instead of the Sequential model (e.g.

```
model = Sequential()
model.add(Dense(32, activation='relu', input_shape=(32,))
model.add(Dense(16, activation='relu'))
model.compile()
```

Should I change my code to using the Sequential model if trying to use `memory_saving_gradients`?

**Comment by gitrdonator** [comment]
@yaroslavvb FYI, here is what the cam map looks like, having changed nothing but the keras backend from `from tensorflow.python.keras._impl.keras import backend as K` to `from keras import backend as K`:

![Figure_1](https://user-images.githubusercontent.com/44979155/60853919-a6d7ea80-a1ed-11e9-9cdd-712487fdafaa.png)
